In [1]:
import pandas as pd
import lightgbm as lgb
from sklearn.metrics import (
    accuracy_score,
    recall_score,
    f1_score,
    precision_score,
    confusion_matrix,
    ConfusionMatrixDisplay,
    roc_auc_score,
    classification_report
)
import matplotlib.pyplot as plt

feature_dataset = pd.read_csv("../data/processed/feature_dataset.csv")
X = feature_dataset.drop(columns=["SepsisLabel"])
y = feature_dataset["SepsisLabel"]

from sklearn.model_selection import train_test_split
X_train, X_test, y_train , y_test = train_test_split(
    X,
    y,
    test_size = 0.2,
    stratify = y,
)

model = lgb.LGBMClassifier(
    objective = "binary",
    random_state = 42
)
model.fit(X_train, y_train)

y_pred = model.predict(X_test)
print(y_pred.shape)


[LightGBM] [Info] Number of positive: 1432, number of negative: 14836
[LightGBM] [Info] Auto-choosing col-wise multi-threading, the overhead of testing was 0.010027 seconds.
You can set `force_col_wise=true` to remove the overhead.
[LightGBM] [Info] Total Bins 31146
[LightGBM] [Info] Number of data points in the train set: 16268, number of used features: 192
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.088026 -> initscore=-2.337985
[LightGBM] [Info] Start training from score -2.337985
(4068,)


In [5]:
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test, y_pred)
print(cm)
from sklearn.metrics import (
    accuracy_score,
    precision_score,
    recall_score,
    f1_score,
    roc_auc_score,
)

print("Accuracy :", accuracy_score(y_test, y_pred))
print("Precision:", precision_score(y_test, y_pred))
print("Recall   :", recall_score(y_test, y_pred))
print("F1 Score :", f1_score(y_test, y_pred))
y_prob = model.predict_proba(X_test)[:, 1]

auc = roc_auc_score(y_test, y_prob)

print("AUROC:", auc)



[[3693   17]
 [ 125  233]]
Accuracy : 0.9650934119960669
Precision: 0.932
Recall   : 0.6508379888268156
F1 Score : 0.7664473684210527
AUROC: 0.952699182339743


TOP RIGHT(3693):in this confusion matrix the top left represent the true negative mean by if the person doesnt have sepsis and model predicted crt no sepsis : 3693
TOP RIGHT (17): false positive (false alarm): actually the patient doesnt have sepsis but model predicted as sepsis
BOTTOM LEFT (125): false negative: actually the person have sepsis but model predicted as no sepsis
BOTTOM RIGHT(233): true positive actually the person have sepsis and the model predicted as sepsis